src/dpx.imageio/dpxoutput.cpp

```diff
@@ -370,8 +370,7 @@ DPXOutput::open(const std::string& name, const ImageSpec& userspec,
 
     ParamValue* kc = spec0.find_attribute("smpte:KeyCode", TypeKeyCode, false);
     if (kc) {
-        int* array = (int*)kc->data();
-        set_keycode_values(array);
+        set_keycode_values(cspan<int>((int*)kc->data(), 7));
```

Comment:
How sure are we that there's actually 7 ints worth of data to walk through?

Reply:
Because the find() will only succeed for an item of TypeKeyCode, which is represented by 7 ints.

Reply:
Ah yes, find does end up checking the type as well. Alright.